pkg/rotator/rotator.go

```diff
@@ -191,13 +202,15 @@ func addNamespacedCache(mgr manager.Manager, namespace string) (cache.Cache, err
 	if err != nil {
 		return nil, err
 	}
-	if err := mgr.Add(c); err != nil {
+	// Wrapping the cache to make sure it's also started when the manager
+	// hasn't been leader elected and CertRotator.RequireLeaderElection is false.
+	if err := mgr.Add(&cacheWrapper{Cache: c, needLeaderElection: cr.RequireLeaderElection}); err != nil {
```

Comment:
where do we check if CertRotator.RequireLeaderElection is false?

Reply:
It is ultimately checked by controller-runtime, that calls the `NeedLeaderElection` method here:

https://github.com/kubernetes-sigs/controller-runtime/blob/1b80b9629af8ec55ae200e2c649a9603533eed20/pkg/manager/runnable_group.go#L66

This is the same mechanism used to control how the other runnables, i.e., `CertRotator` and `ReconcileWH`, start w.r.t. leader election.